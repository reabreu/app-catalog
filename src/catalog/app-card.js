import React from "react";
import {
  Li,
  AppLink,
  AppIcon,
  AppTitle,
  AppVersion,
  CardWrapper,
} from "./styles";

export const AppCard = ({ app }) => {
  return (
    <Li key={app.id}>
      <AppLink to={`/app/${app.id}`}>
        <CardWrapper>
          <AppIcon src={app.iconURL} alt={app.name} />
          <div>
            <AppTitle>{app.name}</AppTitle>
            <AppVersion>{app.version}</AppVersion>
          </div>
        </CardWrapper>
      </AppLink>
    </Li>
  );
};
