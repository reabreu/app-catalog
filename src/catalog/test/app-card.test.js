import React from "react";
import { render } from "@testing-library/react";
import { AppCard } from "../app-card";
import { BrowserRouter as Router } from "react-router-dom";

describe("<AppCard /> spec", () => {
  it("Renders Card", () => {
    const appData = {
      id: "foo",
      iconURL: "foo.jpg",
      name: "foo app",
      version: "1.2.2",
    };

    const { asFragment } = render(
      <Router>
        <AppCard app={appData} />
      </Router>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
