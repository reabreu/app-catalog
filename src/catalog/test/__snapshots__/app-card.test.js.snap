// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppCard /> spec Renders Card 1`] = `
<DocumentFragment>
  <li
    class="sc-AxiKw bqFWkx"
  >
    <a
      class="sc-Axmtr cIJKbp"
      href="/app/foo"
    >
      <div
        class="sc-AxhCb kqkZWI"
      >
        <img
          alt="foo app"
          class="sc-AxhUy kfGxSb"
          src="foo.jpg"
        />
        <div>
          <p
            class="sc-AxgMl kdnRNu"
          >
            foo app
          </p>
          <p
            class="sc-AxheI gyIMnE"
          >
            1.2.2
          </p>
        </div>
      </div>
    </a>
  </li>
</DocumentFragment>
`;
